### PR TITLE
✓ CRC32 Implementation

### PR DESCRIPTION
**Pull request by calebcordry**
Adds a standard (?) CRC32 implementation. Includes some test cases that should be easily copy & paste-able for analytics vendors.

### CODE REVIEW

File: extensions/amp-analytics/0.1/crc32.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  }
+  return crcTable;
+}
+
+/**
+ * Returns CRC32 checksum for provided string.
+ * @param {string} str
+ * @return {number}
+ */
+export function crc32(str) {
```
Comment (lannka) [inline extensions/amp-analytics/0.1/crc32.js]: public method first

File: extensions/amp-analytics/0.1/crc32.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+/**
+ * Standard key for CRC32
+ * https://en.wikipedia.org/wiki/Cyclic_redundancy_check#Polynomial_representations_of_cyclic_redundancy_checks
+ * @const {number}
+ */
+const CRC32_KEY = 0xEDB88320;
+
+/** @const {Array<number>} */
+const CRC_TABLE = makeCrcTable();
```
Comment (lannka) [inline extensions/amp-analytics/0.1/crc32.js]: can we lazily initialize the table? we should not do this work at runtime start up time

File: extensions/amp-analytics/0.1/crc32.js (hunk 1 above)
Comment (calebcordry) [reply extensions/amp-analytics/0.1/crc32.js]: Done.

File: extensions/amp-analytics/0.1/crc32.js (hunk 2 above)
Comment (calebcordry) [reply extensions/amp-analytics/0.1/crc32.js]: Done.

File: extensions/amp-analytics/0.1/test/test-crc32.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    input: 'helloworld',
+    output: 4192936109,
+  },
+  {
+    input: '12345',
+    output: 3421846044,
+  },
+  {
+    input: '漢字',
+    output: 325357704,
```
Comment (jridgewell) [inline extensions/amp-analytics/0.1/test/test-crc32.js]: I don't think this is correct. CRC32 operates on bytes, not (UTF16) code points (Javascript strings).

You're going to have to convert from UTF16 code points to bytes, like https://github.com/SheetJS/js-crc32/blob/58bae10e50cdbbb945007baf1c0ca981756fbc13/crc32.js#L91-L107.

Comment (jridgewell) [reply extensions/amp-analytics/0.1/test/test-crc32.js]: Or, you can use our clever [`utf8Encode`](https://github.com/ampproject/amphtml/blob/464d59514d49f0b2cd08fdfc983beca83383a3c5/src/utils/bytes.js#L32-L42)

**Pull request merged**